glib.yaml

```diff
@@ -0,0 +1,99 @@
+package:
+  name: glib
+  version: 2.74.3
+  epoch: 0
+  description: Common C routines used by Gtk+ and other libs
+  target-architecture:
+    - all
+  copyright:
+    - paths:
+        - "*"
+      attestation: TODO
+      license: LGPL-2.1-or-later
+environment:
+  contents:
+    packages:
+      - busybox
+      - ca-certificates-bundle
+      - build-base
+      - automake
+      - autoconf
+      - bzip2-dev
+      - gettext-dev
+      - libxml2-utils
+      - libxslt
+      - python3
+      - pcre2-dev
+      - meson
+      - zlib-dev
+      - libffi-dev
+      - util-linux-dev
+      - libxslt-dev
+      - python3-dev
+      - libmount
+pipeline:
+  - uses: fetch
+    with:
+      expected-sha256: e9bc41ecd9690d9bc6a970cc7380119b828e5b6a4b16c393c638b3dc2b87cbcb
+      uri: https://download.gnome.org/sources/glib/2.74/glib-${{package.version}}.tar.xz
+  - uses: meson/configure
+    with:
+      opts: |
+        --default-library=both \
+        -Dman=false \
+        -Dtests=false
+  - uses: meson/compile
+  - uses: meson/install
+  - uses: strip
+subpackages:
+  - name: glib-static
+    pipeline:
+      - uses: split/static
+    dependencies:
+      runtime:
+        - gettext-static
+    description: glib static
+  - name: glib-dev
+    pipeline:
+      - uses: split/dev
+      - runs: |
+          mkdir -p ${{targets.subpkgdir}}/usr/bin
+          mkdir -p ${{targets.subpkgdir}}/usr/share
+
+
+          find ${{targets.destdir}}/usr/bin ! -name "glib-compile-schemas" -a \( \
+              -name "gdbus-codegen" -o \
+              -name "gobject-query" -o \
+              -name "gresource" -o \
+              -name "gtester*" -o \
+              -name "glib-*" \) \
+              -exec mv {} ${{targets.subpkgdir}}/usr/bin \;
+
+          mv ${{targets.destdir}}/usr/share/gdb ${{targets.subpkgdir}}/usr/share/
+          mv ${{targets.destdir}}/usr/share/glib-2.0 ${{targets.subpkgdir}}/usr/share/
+    dependencies:
+      runtime:
+        - glib
+        - bzip2-dev
+        - gettext-dev
+        - libxml2-utils
+        - libxslt
+        - python3
+        - libffi-dev # needed by gobject-2.0 when building gobject-introspection
+        - pcre2-dev # needed by glib-2.0 when building gobject-introspection
+        - util-linux-dev # needed by gio-2.0 when building gobject-introspection
+
+      # TODO review if this is needed
```

Comment:
Left as a todo, I've got a bunch more PRs to follow so would rather get something working in first if possible